data-reconciliation-app/src/models/summary-record.ts

```diff
@@ -0,0 +1,145 @@
+
+export enum SummaryGroupStatus {
+  NotEnoughVotes = "NOT_ENOUGH_VOTES",
+  LackOfConsensus = "LACK_OF_CONSENSUS",
+  InConsensus = "IN_CONSENSUS",
+}
+
+export enum SummaryStatus {
+  Majority = "Majority",
+  Minority = "Minority",
+}
+
+export interface SummaryRecordProps {
+  key: string;
+  value: DataAttributeType;
+  type: string;
+  status: SummaryStatus;
+  groupStatus: SummaryGroupStatus;
+  membersInAgreementCount: number;
+  membersInDisagreementCount: number;
+  votesCount: number;
+  uniqueValuesCount: number;
+}
+
+export class SummaryRecord implements SummaryRecordProps {
+  readonly key: string;
+  readonly value: DataAttributeType;
+  readonly type: string;
+  readonly status: SummaryStatus;
+  readonly groupStatus: SummaryGroupStatus;
+  readonly membersInAgreementCount: number;
+  readonly membersInDisagreementCount: number;
+  readonly activeMembersCount: number;
+  readonly votesCount: number;
+  readonly uniqueValuesCount: number;
+
+  private constructor(summaryRecord: SummaryRecordProps) {
+    this.key = summaryRecord.key;
+    this.value = summaryRecord.value;
+    this.type = summaryRecord.type;
+    this.status = summaryRecord.status;
+    this.groupStatus = summaryRecord.groupStatus;
+    this.membersInAgreementCount = summaryRecord.membersInAgreementCount;
+    this.membersInDisagreementCount = summaryRecord.membersInDisagreementCount;
+    this.votesCount = summaryRecord.votesCount;
+    this.uniqueValuesCount = summaryRecord.uniqueValuesCount;
+  }
+
+  public static create(
+    memberId: string,
+    record: ReconciledRecord
+  ): ServiceResult<SummaryRecord> {
+    if (!record.key) {
+      return ServiceResult.Failed({
+        errorMessage: "Error: key cannot be null or empty",
+        errorType: "InvalidRecordKey",
+      });
+    }
+
+    if (!record.values) {
+      return ServiceResult.Failed({
+        errorMessage: "Error: values can not be null or empty",
+        errorType: "InvalidRecordValue",
+      });
+    }
+
+    if (!record.values.hasOwnProperty(memberId)) {
+      return ServiceResult.Failed({
+        errorMessage: "Error: The key does not exist",
+        errorType: "InvalidRecordKey",
+      });
+    }
+
+    // list of all members' Ids
+    const memberIds = Object.keys(record.values);
+
+    // list of all ingested values for same key
+    const values = Object.values(record.values);
+
+    // create a list of unique Values using set data structure
+    const uniqueValues = new Set(values);
+
+    // current member vote
+    const memberValue = record.values[memberId];
+
+    // count of members who voted
+    const votesCount = memberIds.length;
+
+    // count of members who submitted a value same as the current member (caller)
+    const membersInAgreementCount = memberIds.filter(
+      (key) => record.values[key] == memberValue
+    ).length;
+
+    // count of members who submitted a value differ from the current member (caller)
```

Comment:
thank you for adding comments on how we are generating each field...very readable!